server/src/main/java/org/opensearch/index/SegmentReplicationShardStats.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.unit.ByteSizeValue;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * SegRep stats for a single shard.
+ *
+ * @opensearch.internal
+ */
+public class SegmentReplicationShardStats implements Writeable, ToXContentFragment {
+    private final String allocationId;
```

Comment:
I think allocationId is volatile and changes with shard allocation, failover or relocations. If we want to track this over indexShard lifecycle (which I believe we want), do we want to track this with Index shard Id ?

Reply:
The allocation id is unique to the node + shard placement. If it is changed with one of those events & becomes out of sync, the primary will automatically update to reflect the change - this logic is all in ReplicationTracker, which is why I chose to reuse that class vs move this tracking to a separate component.  ShardId is used one level higher to identify the whole replication group.

Reply:
With that said.. returning it here is useless bc I'm not sure we surface allocationID in any other api...Will swap this to either node/ip

Reply:
Updated to swap out allocationId with the shard's current node ID.

Reply:
@dreamer-89 I like this better with node ID, but I was wrong you can fetch allocation_Id from cluster stats api.